app/views/hiring_staff/vacancies/vacancy_form_partials/_submit.html.haml

```diff
@@ -2,4 +2,5 @@
   = f.govuk_submit t('buttons.update_job'), classes: 'update-listing-gtm'
 - else
   = f.govuk_submit classes: 'govuk-!-margin-bottom-5 save-listing-gtm'
-  = f.govuk_submit t('buttons.save_and_return_later'), secondary: true, classes: 'button-link save-and-return-listing-gtm'
+  - unless @job_location_form.present? # || the other form
```

Comment:
I'd prefer to the keep the partial as simple as possible, and make the distinction in the view for the job_location_form instead - if that makes sense?